pyvrp/stop/StoppingCriterion.py

```diff
@@ -6,6 +6,19 @@ class StoppingCriterion(Protocol):  # pragma: no cover
     Protocol that stopping criteria must implement.
     """
 
+    def fraction_remaining(self) -> float | None:
```

Comment:
Maybe rename to `progress()`, where 0 means no progress and 1 means that the stopping criterion has been met?